multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -239,6 +262,21 @@ def fit(
                 raise NotImplementedError(
                     f"Type of the column is not supported currently. Received {col_name}={col_type}."
                 )
+
+    def fit_y(self, y: pd.Series):
```

Comment:
One caveat is that `fit_x` and `fit_y` may better be combined still. We can allow Y=None in `fit()`. This aligns with the design of sklearn.

However, I'm approving the PR now because I feel that we can refactor the code in a later PR.

Reply:
Added back  `fit()`, which calls `_fit_x()` and `_fit_y()` internally.